src/tools/crawler/main.rs

```diff
@@ -147,7 +153,13 @@ async fn main() {
                 }
             }
 
-            thread::sleep(Duration::from_secs(30));
+            if start_time.elapsed() >= Duration::from_secs(SUMMARY_LOOP_INTERVAL) {
+                warn!(parent: crawler.node().span(), "summary calculation took more time than the loop interval");
+            }
+
+            thread::sleep(
+                Duration::from_secs(SUMMARY_LOOP_INTERVAL).saturating_sub(start_time.elapsed()),
+            );
```

Comment:
we should bind the result of the subtraction to some new variable instead of using `elapsed` twice here